filelog.go

```diff
@@ -316,13 +411,18 @@ func (w *FileLogWriter) handleRotate(rotateTime time.Time) error {
 			w.closeLogFile()
 
 			// Rename the file to its newfound home
-			err = os.Rename(w.filename, fname)
+			err = os.Rename(w.filename, rotatedName)
 			if err != nil {
 				return fmt.Errorf("Rotate: %s\n", err)
 			}
 		}
 	}
 
+	// If we're configured to archive files, signal the background goroutine
+	if w.filesToKeep > 0 {
```

Comment:
`rotatedName` is declared outside the `if`, so it isn't empty. There was a much bigger piece of logic here before, but I'll move it inside `w.rotate` now.